extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/handlers/DevClasspathStaticHandler.java

```diff
@@ -0,0 +1,175 @@
+package io.quarkus.vertx.http.runtime.handlers;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.util.Set;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.vertx.http.runtime.GeneratedStaticResourcesRecorder;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.impl.MimeMapping;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.StaticHandler;
+
+/**
+ * {@link StaticHandler} implementation to handle static resources using the Classpath.
+ * This is meant to be used on {@link io.quarkus.runtime.LaunchMode#DEVELOPMENT} mode.
+ */
+public class DevClasspathStaticHandler implements Handler<RoutingContext> {
+
+    private static final Logger LOG = Logger.getLogger(DevClasspathStaticHandler.class);
+    private static final int HTTP_STATUS_OK = 200;
+    private static final int HTTP_STATUS_NO_CONTENT = 204;
+    private static final String ALLOW_HEADER = "Allow";
+    private static final String ALLOW_HEADER_VALUE = "HEAD,GET,OPTIONS";
+    private final Set<String> generatedResources;
+    private final Set<String> compressedMediaTypes;
+    private final ClassLoader currentClassLoader;
+    private final boolean enableCompression;
+    private final String indexPage;
+    private final Charset defaultEncoding;
+
+    public DevClasspathStaticHandler(Set<String> generatedResources, DevClasspathStaticHandlerOptions options) {
+        this.generatedResources = generatedResources;
+        this.compressedMediaTypes = options.getCompressMediaTypes();
+        this.currentClassLoader = Thread.currentThread().getContextClassLoader();
+        this.enableCompression = options.isEnableCompression();
+        this.indexPage = options.getIndexPage();
+        this.defaultEncoding = options.getDefaultEncoding();
+    }
+
+    @Override
+    public void handle(RoutingContext context) {
+
+        String resolvedPath = resolvePath(context);
+        String path = resolvedPath.endsWith("/") ? resolvedPath.concat(this.indexPage) : resolvedPath;
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debugf("Handling request for path '%s'", path);
+        }
+
+        boolean containsGeneratedResource = this.generatedResources.contains(path);
+
+        if (!containsGeneratedResource) {
+            beforeNextHandler(this.currentClassLoader, context);
+            return;
+        }
+
+        if (context.request().method().equals(HttpMethod.OPTIONS)) {
+            context.response().putHeader(ALLOW_HEADER, ALLOW_HEADER_VALUE).setStatusCode(HTTP_STATUS_NO_CONTENT)
+                    .send();
+            return;
+        }
+
+        compressIfNeeded(context, path);
+
+        context.vertx().executeBlocking(future -> {
```

Comment:
I would object that this is using blocking reads for reading what is most likely a local file, while `StaticHandlerImpl` uses the non-blocking file reading…
But this is only for DEV mode, so… Feels weird though having to resort to a worker thread just to read a file in DEV mode…

Reply:
Yes, this is true, maybe we can know if this is a local file and do a different processing in that case?

Reply:
anyway, I will make a few change in the next few days (to add `Path` support to GeneratedStaticResourceProcessor), this might be our solution.